x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java

```diff
@@ -97,6 +100,21 @@ void doExecute(ActionType<Response> action, Request request, ActionListener<Resp
             }
         };
         manager = SecurityIndexManager.buildSecurityMainIndexManager(client, clusterService);
+
+    }
+
+    public void testIndexWithFaultyMappingOnDisk() {
+        SecurityIndexManager.State state = new SecurityIndexManager.State(randomBoolean() ? Instant.now() : null, true, randomBoolean(),
+                false, null, "not_important", null, null);
+        Runnable runnable = mock(Runnable.class);
+        manager = new SecurityIndexManager(mock(Client.class), RestrictedIndicesNames.SECURITY_MAIN_ALIAS,
+                RestrictedIndicesNames.INTERNAL_SECURITY_MAIN_INDEX_7, SecurityIndexManager.INTERNAL_MAIN_INDEX_FORMAT, () -> {
+                    throw new RuntimeException();
+                }, state);
```

Comment:
Can you extract the `Supplier` into a local so that the variable name/type makes it more obvious what it is?
As it reads now, you have to jump to the constructor to work out what the 5th argument is, and why throwing an exception there achieves the purpose fo this test.